packages/lib/src/vis/shared/AxisSystemProvider.tsx

```diff
@@ -55,46 +48,43 @@ function AxisSystemProvider(props: PropsWithChildren<Props>) {
     [camera, cameraToHtmlMatrix]
   );
 
-  const registeredKeys = Object.values(interactionParams)
-    .map((params) => (params === true ? undefined : params.modifierKey))
-    .filter(isDefined);
-  const keySet = new Set(registeredKeys);
-  if (keySet.size !== registeredKeys.length) {
-    throw new Error('Two interactions were registered on the same key !');
-  }
-
-  const getModifierKey = useCallback(
-    (id: string) => {
-      if (!Object.keys(interactionParams).includes(id)) {
-        throw new Error(`Interaction ${id} was not registered in VisCanvas.`);
-      }
-
-      const params = interactionParams[id];
-
-      return params === true ? undefined : params.modifierKey;
-    },
-    [interactionParams]
-  );
+  const [interactionMap] = useState(new Map<string, Interaction>());
 
   const shouldInteract = useCallback(
     (id: string, event: MouseEvent) => {
-      const params = interactionParams[id];
-
-      if (params !== true && params.disabled) {
+      const registeredKeys = [...interactionMap.values()]
+        .map((params: Interaction) => params.modifierKey)
+        .filter(isDefined);
+
+      const params = interactionMap.get(id);
+      assertDefined(
+        params,
+        `Interaction ${id} was not registered in VisCanvas.`
```

Comment:
I would use present tense here and drop `in VisCanvas`: `Interaction ${id} is not registered`.